File: Winter2022/lab-solutions/java/Lab7/RoomPersistenceLabSkeleton/app/src/main/java/mobiledev/unb/ca/roompersistencelab/ui/ItemViewModel.java
package mobiledev.unb.ca.roompersistencelab.ui;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;

import java.util.List;

import mobiledev.unb.ca.roompersistencelab.entity.Item;
import mobiledev.unb.ca.roompersistencelab.repository.ItemRepository;

public class ItemViewModel extends AndroidViewModel {
    private final ItemRepository itemRepository;

    public ItemViewModel(@NonNull Application application) {
        super(application);
        itemRepository = new ItemRepository(application);
    }

    // TODO
    //  Add mapping calls between the UI and Database
    public void insert(String name, int num) {
        itemRepository.insertRecord(name, num);
    }

    public List<Item> findItemsByName(String name) {
        return itemRepository.findItemsWithName(name);
    }
}

File: Winter2022/labs/kotlin/Lab7/RoomPersistenceLabSkeleton/app/src/main/java/mobiledev/unb/ca/roompersistencelab/ui/ItemViewModel.kt
package mobiledev.unb.ca.roompersistencelab.ui

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import mobiledev.unb.ca.roompersistencelab.repository.ItemRepository

class ItemViewModel(application: Application) : AndroidViewModel(application) {
    private val itemRepository: ItemRepository

    init {
        itemRepository = ItemRepository(application)
    }

    // TODO
    //  Add mapping calls between the UI and Database
}